lib/mapping/CCampaignHandler.cpp

```diff
@@ -35,27 +36,73 @@
 
 VCMI_LIB_NAMESPACE_BEGIN
 
+CampaignRegions::RegionDescription CampaignRegions::RegionDescription::fromJson(const JsonNode & node)
+{
+	CampaignRegions::RegionDescription rd;
+	rd.infix = node["infix"].String();
+	rd.xpos = static_cast<int>(node["x"].Float());
+	rd.ypos = static_cast<int>(node["y"].Float());
+	return rd;
+}
+
+CampaignRegions CampaignRegions::fromJson(const JsonNode & node)
+{
+	CampaignRegions cr;
+	cr.campPrefix = node["prefix"].String();

```

Comment:
Personally I would've used full names in config instead of runtime generation, at least for vcmi campaings.
Not critical though